pkg/finalizer/finalizer_test.go

```diff
@@ -0,0 +1,92 @@
+package finalizer
+
+import (
+	"context"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest/printer"
+)
+
+type mockFinalizer struct{}
+
+func (f mockFinalizer) Finalize(context.Context, client.Object) (needsUpdate bool, err error) {
+	return true, nil
+}
+func TestFinalizer(t *testing.T) {
+	RegisterFailHandler(Fail)
+	suiteName := "Finalizer Suite"
+	RunSpecsWithDefaultAndCustomReporters(t, suiteName, []Reporter{printer.NewlineReporter{}, printer.NewProwReporter(suiteName)})
+}
+
+var _ = Describe("TestFinalizer", func() {
+	var err error
+	var pod *corev1.Pod
+	var finalizers Finalizers
+	var f mockFinalizer
+	BeforeEach(func() {
+		pod = &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Finalizers: []string{"finalizers.sigs.k8s.io/testfinalizer"},
+			},
+		}
+		Expect(pod).NotTo(BeNil())
+
+		finalizers = NewFinalizers()
+		Expect(finalizers).NotTo(BeNil())
+
+		f := mockFinalizer{}
+		Expect(f).NotTo(BeNil())
+
+	})
+	Describe("Finalizer helper library", func() {
```

Comment:
Nit:
```suggestion
	Describe("Register", func() {
```

And then put just the Register tests in this block. Ditto for `Finalize`

And then the `It` tests can be `It("successfully registers a finalizer")` to make it read more natually.